site/learn/expert-advice/authentication/webauthn-explained.md

```diff
@@ -0,0 +1,515 @@
+---
+layout: advice
+title: WebAuthn Explained
+description: An in-depth look at how WebAuthn can help you add biometric authentication to your website.
+author: Dan Moore
+image: advice/vendor-lockin/expert-advice-avoiding-authentication-system-lock-in-header-image.png
+category: Authentication
+date: 2022-09-28
+dateModified: 2022-09-28
+---
+
+WebAuthn is a new way to authenticate your users using biometric, secure authentication methods. WebAuthn requires something called an authenticator. What is an authenticator? An authenticator is a hardware or software device that is used to authenticate a user, via biometric or other methods, and that follows certain standards.
+
+That's a bit tautological, isn't it? Well, yes, but read on for more details. Also, clowns.
+
+## WebAuthn Overview
+
+Let's start off with an overview of the use cases, "ceremonies", and types of authenticators that all play a role in a WebAuthn implementation.
+
+There are three main use cases that WebAuthn works well for.
+
+1. Re-authentication, where the user has an account created in another way, but wants an easier way to login.
+1. First factor authentication, where the user has no account, but uses WebAuthn to login instead of a more typical first factor like a username/password combination.
+1. As another factor available for MFA.
+
+The main focus of this article is re-authentication, though the other use cases will be covered briefly.
+
+WebAuthn has flows of data and interaction, termed "ceremonies". These extend a network protocol to include things in the real world, such as physical hardware or human beings. There are two main ceremonies:
+
+1. Authentication, where the user logs in, using WebAuthn
+1. Registration, where the user associates an authenticator with their account
+
+What does an authentication ceremony look like? At a high level, the flow of interaction is:
+
+1. The user tries to login to a website
+1. The website prompts the user to authenticate
+1. The user authenticates against their authenticator
+1. The authenticator signs and passes a response back to the website through the browser
+1. The website verifies the authenticator response
+1. The user is logged in
+
+Whenever the user authenticates against the relevant authenticator, that is an "authorization gesture". Even though "gesture" implies movement, it could be a biometric means like Touch ID, using a PIN, or any other method that an authenticator offers.
```

Comment:
An [authorization gesture](https://www.w3.org/TR/webauthn-2/#authorization-gesture) does involve some sort of physical interaction. This is an important piece of WebAuthn as it requires a user to be present with the device. For example, I would not be able to use an authenticator by remotely taking over someone's machine.